### PR TITLE
Exposed minificationFilter and magnificationFilter for Material 

### PR DESCRIPTION
**Pull request by santilland**
Ok, so here goes first try at making a pull request, quite some text to read to get to know the workflow.
I signed the CLA.
This pull request is related to discussion here:
https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/cesium-dev/CacSIMIhqeo/jdpgL-v9AQAJ
and to ticket #8328 
I basically tried to mimic/copy the approach taken for the imagerylayer.
A sandcastle example:

```
var viewer = new Cesium.Viewer('cesiumContainer');

var checker2x2 = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAIAAAACCAIAAAD91JpzAAAACXBIWXMAAC4jAAAuIwF4pT92AAAAB3RJTUUH4wwSDS0atnSBXwAAABl0RVh0Q29tbWVudABDcmVhdGVkIHdpdGggR0lNUFeBDhcAAAAVSURBVAjXBcEBAQAAAIAQ/08XUKkwKeQF+2OdNgAAAAAASUVORK5CYII=';

var rectanglelinear = viewer.scene.primitives.add(new Cesium.Primitive({
    geometryInstances : new Cesium.GeometryInstance({
        geometry : new Cesium.RectangleGeometry({
            rectangle : Cesium.Rectangle.fromDegrees(-140.0, 10.0, -80.0, 40.0),
            vertexFormat : Cesium.EllipsoidSurfaceAppearance.VERTEX_FORMAT
        })
    }),
    appearance : new Cesium.EllipsoidSurfaceAppearance({
        aboveGround : false
    })
}));

rectanglelinear.appearance.material = new Cesium.Material({
    fabric : {
        type : 'DiffuseMap',
        uniforms : {
            image : checker2x2
        }
    }
});

var rectanglenearest = viewer.scene.primitives.add(new Cesium.Primitive({
    geometryInstances : new Cesium.GeometryInstance({
        geometry : new Cesium.RectangleGeometry({
            rectangle : Cesium.Rectangle.fromDegrees(-80.0, 10.0, -20.0, 40.0),
            vertexFormat : Cesium.EllipsoidSurfaceAppearance.VERTEX_FORMAT
        })
    }),
    appearance : new Cesium.EllipsoidSurfaceAppearance({
        aboveGround : false
    })
}));

rectanglenearest.appearance.material = new Cesium.Material({
    fabric : {
        type : 'DiffuseMap',
        uniforms : {
            image : checker2x2
        }
    },
    minificationFilter: Cesium.TextureMinificationFilter.NEAREST,
    magnificationFilter: Cesium.TextureMagnificationFilter.NEAREST
});
```
Should look like this:
minmagexample

On the left the current default linear approach on the right using nearest neighbor, really useful for low resolution images/materials.

Hope i did not forget anything, let me know if you need something more,
kind regards,
Daniel

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @santilland!
* :x: Missing CLA.
    * Please send in a [Contributor License Agreement](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla) (CLA) and comment back here to let us know to check this!
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/materialminmag/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by OmarShehata** [comment]
Thanks @santilland! I can confirm we just received your signed corporate CLA.

**Comment by IanLilleyT** [comment]
Thanks for the PR @santilland! I made some changes and tests, and this should be ready after another quick review.

**Comment by lilleyse** [comment]
I pushed a fix for video materials.

sampling

[Local sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=3VZbbxo5FP4rFn2ASMST9KKNKEGbNqSKlGxXJKp2JaTKjM8MVj321D6G0FX++x7PBQbQbvelD9t5APtcvu9cGVbCsZWCNTh2yQys2XvwKhT8UyUb9NPq+t4aFMqA6w/ZX3PD6PFLu56BkeDurC2nzlnn2YhlQnsYbk2ClldGFQKBdOgCzM3zydu5mZtVRSzBTjUUYJDopU1DPPIcsJG+29zKQR+dUJrId54OUhQm16ApKhFjr5PgPgUDvHSqUKhW4LmQctDJ6/dWM2jyyMEWgG5zazwhUroUaMf+w4G6deu67nvM2tBa165LfLaxk9+hD8+cLa4hdwB+cHr++oyfDdl59Xl6UX1F0clwH3EFDuHpxjoq9A50qrUqvVXyIbhMpHBVllSqmAT/NJ09Tv/4fPNxdn/1uMN6PqnPzy2B2Lrs5/jP0N1kxcKu4IOzwch2Mlr8OAZ1Nw86yXeUPI6NU0LvT+Z9I22ZMrFwKiWCDjNuyhhx/1plWfBwL8p+p2TBqIxq5fd94kOTmnfa0lLxa8hE0Hgb1beyU7Amn+1U/4dseEvPa7bLvTU4nvCIAx5/0hG/6E74y597wptO/j9HvAEvFCGrVKCy5kZpct5CPcITBgf3Rxb8t+nVbPrw2EKI/LsYxyYtyPGq/UtZv7trvWFv7HGjYVLH9qsqSuuQBacHnCcIRakJyyeLkH4B5Kn3NXs0ftG8l7olLq1XMegRzYa3OiC83SkXFtEWI/bLm/KpI3YqX+KInXVEayVxOWKvXp7tWS6hNj2/2MmpQeOkzYHOUq2Ykpfz3sGre95jqRbekyYLWj+obzDvTcYJ2U+6btoKqUz+kVZOi000WZ5P7moh53yc0PXYC63VC+E6iKStit3o61JREAXVJEroKAJaqu+mvmn6H1GfUme9t1QWZRoBJeGs9tWtadXY2+Bobb1LSbxELP0oSbwwMhUe6Xemzp+ntkgeBPURrgWKpIqJ+qny09hT+jBm87kJj69hURBfXC4CrWyTWvYjWYvy9SFpJfqhnHZ1yPk1qPQLqgK2zH8SMVs4u/Y05dLSm8NYZD6U1ZLgEtg4tRImlf84qc4M6u3icS4rBYH9DQ) - you need to disable CORS for the video to work locally (I use the [allow-cors-access-control](https://chrome.google.com/webstore/detail/allow-cors-access-control/lhobafahddgcelffkeicbaginigeejlf?hl=en) Chrome extension but only for dev purposes like this, use at your own risk).

File: Source/Scene/Material.js [hunk 1]
```diff
@@ -298,6 +301,28 @@ import when from '../ThirdParty/when.js';
...
+        /**
+         * The {@link TextureMinificationFilter} to apply to this material.
+         * The default value is {@link TextureMinificationFilter.LINEAR}.
+         *
+         * To take effect, this property must be set before the texture is created.
+         *
+         * @type {TextureMinificationFilter}
+         * @default {@link TextureMinificationFilter.LINEAR}
+         */
+        this.minificationFilter = defaultValue(options.minificationFilter, TextureMinificationFilter.LINEAR);
```
Comment (lilleyse) [inline Source/Scene/Material.js]: These options should be documented in the `Material` constructor.

If these options only take effect before textures are created maybe they should only be documented in the constructor and not here.

**Comment by lilleyse** [comment]
Thanks @santilland and @IanLilleyT!

**Comment by cesium-concierge** [comment]
Congratulations on closing the issue! I found these Cesium forum links in the comments above:

https://groups.google.com/forum/?utm_medium&#x3D;email&amp;utm_source&#x3D;footer#!msg/cesium-dev/CacSIMIhqeo/jdpgL-v9AQAJ

If this issue affects any of these threads, please post a comment like the following:

> The issue at https://github.com/AnalyticalGraphicsInc/cesium/pull/8473 has just been closed and may resolve your issue. Look for the change in the next stable release of Cesium or get it now in the master branch on GitHub https://github.com/AnalyticalGraphicsInc/cesium.

**Comment by santilland** [comment]
Great! Thanks for the merge!

**Pull request merged**